ui/client/components/Process/State/StateIcon.tsx

```diff
@@ -0,0 +1,134 @@
+import React from "react"
+import {CSSTransition, SwitchTransition} from "react-transition-group"
+import {withTranslation} from "react-i18next"
+import {WithTranslation} from "react-i18next/src"
+import {compose} from "redux"
+
+import ProcessStateUtils from "./ProcessStateUtils"
+import {ProcessStateType, ProcessType} from "../ProcessTypes"
+import {absoluteBePath} from "../../../common/UrlUtils"
+import {unknownName, unknownTooltip} from "../ProcessMessages"
+
+import {Popover} from "react-bootstrap"
+import {OverlayTrigger} from "react-bootstrap/lib"
+
+type State = {
+  animationTimeout: {
+    enter: number;
+    appear: number;
+    exit: number;
+  };
+}
+
+type OwnProps = {
+  processState?: ProcessStateType;
+  isStateLoaded: boolean;
+  process: ProcessType;
+  animation: boolean;
+  height: number;
+  width: number;
+}
+
+type Props = OwnProps & WithTranslation
+
+class StateIcon extends React.Component<Props, State> {
+  static defaultProps = {
+    isStateLoaded: false,
+    processState: null,
+    animation: true,
+    height: 24,
+    width: 24,
+  }
+
+  // eslint-disable-next-line i18next/no-literal-string
+  static popoverConfigs = {placement: "bottom", triggers: ["click"]}
+
+  state = {
+    animationTimeout: {
+      enter: 500,
+      appear: 500,
+      exit: 500,
+    },
+  }
+
+  animationListener = (node, done) => node.addEventListener("transitionend", done, false)
+
+  getTooltip = (process: ProcessType, processState: ProcessStateType, isStateLoaded: boolean) => {
+    if (isStateLoaded === false) {
+      return process.state?.tooltip || unknownTooltip()
+    }
+
+    return processState?.tooltip || unknownTooltip()
+  }
+
+  getIcon = (process: ProcessType, processState: ProcessStateType, isStateLoaded: boolean) => {
+    if (isStateLoaded === false) {
+      return absoluteBePath(process.state?.icon || ProcessStateUtils.UNKNOWN_ICON)
+    }
+
+    return absoluteBePath(processState?.icon || ProcessStateUtils.UNKNOWN_ICON)
+  }
+
+  popoverOverlay = (name: string, tooltip: string, errors: Array<string>) => {
+    const {t} = this.props
```

Comment:
What is `t`?

Reply:
https://www.i18next.com/overview/api#t :)